onnxruntime/contrib_ops/cpu/moe/moe_utils.cc

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "contrib_ops/cpu/moe/moe_utils.h"
+#include <cmath>
+#include <algorithm>
+
+namespace onnxruntime {
+namespace contrib {
+
+float ApplyActivation(float x, ActivationType activation_type) {
+  switch (activation_type) {
+    case ActivationType::Relu:
+      return std::max(0.0f, x);
+    case ActivationType::Gelu:
+      return 0.5f * x * (1.0f + std::tanh(0.7978845608f * (x + 0.044715f * x * x * x)));
+    case ActivationType::Silu:
+      return x * (1.0f / (1.0f + std::exp(-x)));
+    case ActivationType::Identity:
+      return x;
+    case ActivationType::SwiGLU:
+      // SwiGLU: This is handled specially as it requires gating, not applied here
+      return x;
+    default:
+      return x;  // Default to identity
+  }
+}
+
+void ApplySwiGLU(const float* fc1_output, float* result, int64_t inter_size) {
+  constexpr float swiglu_alpha = 1.702f;
+  for (int64_t i = 0; i < inter_size; ++i) {
+    float linear_val = fc1_output[i];             // First half: linear projection
+    float gate_val = fc1_output[i + inter_size];  // Second half: gate projection
```

Comment:
We shall use interleave format like cuda by default. Interleave might be faster due to cache.